azure-pipelines.yml

```diff
@@ -3,7 +3,7 @@
 jobs:
 - job: Test
   pool:
-    vmImage: Ubuntu 16.04
+    vmImage: 'ubuntu-18.04'
```

Comment:
What about changing to `ubuntu-latest`? Just would mean that we don't have to keep updating the image as older ones become deprecated?

Reply:
I can check that again, but I remember I needed many changes in the code to happy `clang` for that change and I decided to go with the easy-fix for now. Also, some of the issues I could troubleshoot quicker because I had a pointer that the os version differs. If it was set to `ubuntu-latest` I would not have had that pointer.

Reply:
Ok yeah, if it's a problem, it's fine - just something that could become an issue in the future. 
The current version for `ubuntu-latest` is here: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml